Total Remap of the Ranger

Latentish opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->
## About The Pull Request
The old ranger is just about a year old. This totally remaps this ship and rebalances it for our current standards.
![rangerpng](https://github.com/shiptest-ss13/Shiptest/assets/55075747/1290c0e9-5efb-4235-b047-a8993466f5ca)
![2023-08-19 12 35 04](https://github.com/shiptest-ss13/Shiptest/assets/55075747/acfb106e-b770-4f9a-8f76-bdbcb3dee6ff)

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
This is good for the game because it replaces a year old ship with something that's more up to par with our standards.
## Why It's Good For The Game
Ship upkeep and maintaining our maps is a good thing. Plus the old ranger had a bunch of old snowflaked code that was a bandaid fix for the ship access system not existing yet. (It unsnowflakes it).
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl:
tweak: Remaps the Ranger-class
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Latentish committed:
Base

-Adds a basis for the remap

Latentish committed:
Adds most of the ship

still gotta make it pretty

Latentish committed:
Tweaks the Json

-Removes all non lp jobs and adds an assistant.

Latentish committed:
Assorted fixes

-Adds firelocks
-Adds a bit of precharge to the SMES
-Gets the ship to actually spawn (ports are still fucked)

Still needs fire alarms

Latentish committed:
More small fixes

-Changes some flooring in the ai chamber
-Adds missing piping.

Latentish commented:
ToDO:

- [x] Fire Alarms
- [x] Cameras
- [x] Holopads
- [x] Access
- [x] Ripping out old code
- [x] Fixing docking ports
- [x] Decals

BarteG44 commented:
the modern railing handrails can be set to have a custom color 
![image](https://github.com/shiptest-ss13/Shiptest/assets/105491762/d50a528e-9aa0-46e3-bdc4-b68ffad9452e)

Latentish committed:
Merge remote-tracking branch 'upstream/master' into New-Age-Ranger

Latentish committed:
"fixes" ports

Ship still turns, but it flies in the right direction at least.

Latentish committed:
Adds more ship

-Adds some decaling
-Adds more ship

Latentish committed:
decals and medicine

-Gives the medic some gear
-Adds more decal work

Latentish committed:
Unscrungles the ranger code

- Removes the snowflaked ranger accesses
- Trims the mission list
- Adds minor details to the ship
- Removes the commissioner job

Latentish committed:
Merge remote-tracking branch 'upstream/master' into New-Age-Ranger

Latentish committed:
Updates tags and description

Titular

Latentish committed:
Fire alarms!!!!

-Almost forgot these

Latentish committed:
removes second riot tag

Whoobsies

Latentish committed:
DECALS!!!

mINOR DECAL FIX!!!!

Latentish committed:
More updates

-Redoes decals
-adds an exosuit fab circuit to the cyborg pile
-adds a second boris
-gives them more air for the emergency entrance/exit

Latentish committed:
whoops

-one of last commit's changes didn't make it into the commit

Latentish committed:
removes engine windows

The engines stop air now!!!!!!!!!!!!

Latentish commented:
![rangernewest](https://github.com/shiptest-ss13/Shiptest/assets/55075747/9bf4804a-29ca-423d-9a55-68bca4f0f5fd)
![2023-06-27 02 29 30](https://github.com/shiptest-ss13/Shiptest/assets/55075747/454cc2e0-08cc-4880-97a7-ec77c5249d8b)
The group pissing picture took 8 minutes to take you get these in a comment this time.

Erikafox reviewed (dismissed):
Something in here has changed the default captain outfit to be the LP lieutenant outfit.

Latentish commented:
> Something in here has changed the default captain outfit to be the LP lieutenant outfit.

I don't see anything in the code that would imply this. Could you give me more specifics? Do you mean outfit datum? Outfit as in clothes? Did you see this on tm? Was it a full outfit swap?

Latentish commented:
> Something in here has changed the default captain outfit to be the LP lieutenant outfit.

We figured this out, and now know that not properly declaring jobs in jsons can cause this. Funny haha moment. I'll make a pr for this l8r.

OvercastAngel commented:
This remap leaves out the intercoms from the original. While not conventionally useful for the crew, it's very necessary for the AI that the ship comes with!

Latentish committed:
Adds intecomms and bedsheets

-Titular

Latentish committed:
Removes the job subtypes

-Removes all job subtypes and leaves the outfits.

Erikafox commented on code/datums/lp_mission_holotapes.dm:
```diff
@@ -286,33 +214,6 @@
 	DELAY 25
 	"}
 
-/obj/item/disk/holodisk/lp/aid/ripley
-	name = "Aid Mission Ripleys in Aid"
-	desc = "A holodisk containing an aid mission for the LP."
-	preset_image_type = /datum/preset_holoimage/commissioner
-	preset_record_text = {"
-	NAME Commissioner Gorre Donn
-	SAY This mission, should you choose to accept it, revolves around an aid objective. The LP will need to provide aid to local vessels not hostile or syndicate.
-	DELAY 25
-	SAY A well equipped Ripley will help any vessel in hard times. Provide 2 allied or otherwise friendly, non-syndicate vessels with an equipped Ripley exosuit.
-	DELAY 25
-	"}
-
```
I actually like some of these aid missions. They seem conceptually neat.

Erikafox commented on code/modules/jobs/job_types/lp_jobs.dm:
```diff
@@ -1,20 +1,7 @@
-	return get_lplieu_access()
-
-/datum/outfit/job/lp/lieutenant
+/datum/outfit/job/captain/nt/lp_lieutenant
```
outfit datums should go in relevant job files. It makes it easier to tell what they inherit from.

Erikafox commented on _maps/configs/nanotrasen_powerrangers.json:
I'd like it if this and the ship file was changed to "nanotrasen_ranger" to fit in with the schema of faction-class

Erikafox reviewed (dismissed):
![image](https://github.com/shiptest-ss13/Shiptest/assets/94164348/ba7a60ad-f8b4-4c83-93ad-0ad39a55b7a1)
Stray pump. This is not enough atmospherics for a ship of this size, especially on free access.
![image](https://github.com/shiptest-ss13/Shiptest/assets/94164348/e9ca6a1f-7fb8-4936-a7ca-4fb9f163d0aa)
This is a very visually noisy room for not much gain. Consider going lighter on the decals while using alternative tiles (try mono tiles under machinery - tables, with a blue trimline pattern.) I would also put a smartfridge for blood instead of a crate, and an anesthetic closet for their anesthetic. The medkits could perhaps go in the cargo bay to encourage set up.
![image](https://github.com/shiptest-ss13/Shiptest/assets/94164348/e9a10301-4555-4a78-9deb-20ac2e2b6b52)
As an aside, the perspective in this room does not work. The railings are too tall and do not line up with the staircase, while the desk lamp is at the high end of the table, instead of the low end.
![StrongDMM_RXzJXjBCVN](https://github.com/shiptest-ss13/Shiptest/assets/94164348/e9d1a023-4cef-4830-892e-90ca1feb8d5a)
Edge right against the stairs damages it as well.
![StrongDMM_Z7eeRr5Vk3](https://github.com/shiptest-ss13/Shiptest/assets/94164348/d2c8aca7-b87a-4f55-914f-67def8db3db0)
Consider using window firelocks in place of doubled locks
![StrongDMM_EaXfh1k0VV](https://github.com/shiptest-ss13/Shiptest/assets/94164348/5f3fa6e6-1559-42dc-8943-8376d479ec26)
This room feels very dry. There's no indication of *where* you should store the plasma, the cans are just kinda there. No dirt, no decals, just floor and cans.

Sidenote: You can attach accesses to buttons. I encourage experimenting with it a little
![StrongDMM_lA22vz74fw](https://github.com/shiptest-ss13/Shiptest/assets/94164348/d9fec78f-fcb6-44a9-a21d-0b55a1e3a0e1)
Random middle of the hallway autolathe and power control console
![image](https://github.com/shiptest-ss13/Shiptest/assets/94164348/27863a8d-7b3b-4c31-bec1-62b41159115f)
This is another monotile room with a weak decal pattern. Try again, breaking it up with other patterns of tiles and alternative decal methods.
![StrongDMM_aB4oezI5CY](https://github.com/shiptest-ss13/Shiptest/assets/94164348/3aff29db-5e43-4e6e-8438-6538b0bd1088)
I like the flooring pattern here. I do not like the equipment.
Power generation, an AI Core, Telecomms, and Robotics should not all be clumped together in one room. 
You can preset telecomms equipment, look at how the tegus, chronicle, talos, and colossus handle it. If you set the "calibrating" var on the messanging server to 0, it will start functioning.
I personally would prefer that the ship only had a relay, set to link to existing NT commnets, and that the rest of the space was used to more clearly define power generation and the robotics.
![StrongDMM_2rtLgW36nV](https://github.com/shiptest-ss13/Shiptest/assets/94164348/25ceef78-bde5-4e4f-94e9-68101decb20b)
Don't place light switches like this. They look like holofields and that's how we set up holofield switches.
![StrongDMM_eHofEL4XG9](https://github.com/shiptest-ss13/Shiptest/assets/94164348/86a5a68e-00b7-4143-a819-c1948869e970)
This room has the same problem as the other engine room. 
![StrongDMM_UT21ekHgAH](https://github.com/shiptest-ss13/Shiptest/assets/94164348/ca380f86-c336-4d16-b3c1-6df72fd43618)
This room suffers the same perspective issues that the medical area does, but the decalling is better. I would consider using red plasteel instead of plastitanium, or ditching the edge around the room.
I would also remove the .38 hunting from the room, and give them something such as an E-TAR SMG. If you wanted to be bold. You could refactor it to allow for quick battery cell swapping and give it a suitably NT name.

![StrongDMM_AUHg9rjG6z](https://github.com/shiptest-ss13/Shiptest/assets/94164348/ad1813d4-66c2-4d66-806c-9be298f52d50)
You can set up brig doors to work with a timer. Look at how the Vela does it in it's brig.
![StrongDMM_csy0muVajk](https://github.com/shiptest-ss13/Shiptest/assets/94164348/1e673b4b-897f-4907-9cd0-3881b043ed0a)
Any airlock this size needs at least two highcap vents.
![StrongDMM_6Zwf3HbOTP](https://github.com/shiptest-ss13/Shiptest/assets/94164348/9215d46f-169c-4507-8c5c-2d48b20d2421)
This is fine for a quick resupply, but think about how any bulky amount of supplies would be moved.

![StrongDMM_bXYAbmIXAN](https://github.com/shiptest-ss13/Shiptest/assets/94164348/fad83d53-b828-41d9-9408-f751d868718c)
Don't maintain the decal pattern you use for walkways to under equipment.
![StrongDMM_frOSZ97qQy](https://github.com/shiptest-ss13/Shiptest/assets/94164348/51fc13b3-af64-4002-9c42-a96d96e73e7a)
I honestly don't think this room needs the railings. It looks p good. Also stray wallmount in the bottom of the screenshot.
![StrongDMM_M5F1lcmyYe](https://github.com/shiptest-ss13/Shiptest/assets/94164348/a0f81708-8f6e-4de4-83b9-b448dfcd3212)
This will lead to the left (rear) being crushed by some subshuttles. Their dock will land exactly on this dock, and that can cause them to partially land on the hull.
![StrongDMM_FrcyFFqMDR](https://github.com/shiptest-ss13/Shiptest/assets/94164348/7da1417f-eaf9-4e88-be97-645816f40a5f)
I'd swap this for 1 pump - 1 big scrub, an oxycan, an aircan, and a spare pacman.
![StrongDMM_2ukPEzFWEl](https://github.com/shiptest-ss13/Shiptest/assets/94164348/3da7744a-581a-47e7-9275-341390f5ac4c)
Try doing a trek style bridge (Beluga has one)
My advice is to lengthen the ship a little bit and use the extra space to more clearly define your areas.

Latentish committed:
Merge remote-tracking branch 'upstream/master' into New-Age-Ranger

Latentish committed:
More remapping

-Changes a lot.
-Ai/telecomms area
-Bridge
-Commons area
-cargo bay
-maints

Latentish committed:
oopdates map

Adds the rest of the ship?

Latentish committed:
Merge remote-tracking branch 'upstream/master' into New-Age-Ranger

Latentish committed:
Woo boy

-Sanitizes ranger code
-Reduces lp file count
-Sanitizes Ranger json
-Various mapchanges

Latentish committed:
whoopsies

-adds well needed commas

Latentish committed:
Linters moment

-Shiplinters

Latentish commented:
hokay it's ready for the testering crew to look at its testerability

Sly9sky commented:
Mainly with what I've seen so far, the Medical Specialist should get a stretcher if they don't have one already, and the bridge doesn't start out with a helm console, to which I think the foremost console should be replaced with one.

Latentish committed:
oopsiewoopsies

-Fixes missing pipes
-Adds the helm =)
-gives the medic roller beds
-removes a stray closet

Latentish committed:
Literally every time I make a ship

-Adds wiring under the engines and prechargers

Latentish committed:
Minor decal updates for spocky

-happy maintainer chwismas

Latentish commented:
this int test is on some shit

Latentish commented:
![image](https://github.com/shiptest-ss13/Shiptest/assets/55075747/92b48720-8ef4-4cf7-99a6-237be6f9b81b)
This is neither from my map, nor from this latest commit. There is no "library" area on the ranger.

Erikafox commented:
![image](https://github.com/shiptest-ss13/Shiptest/assets/94164348/2cb6afa9-bf1e-4f3d-b187-955f32186e07)
the equipment room needs decals, the other room still has the Bad Decal Pattern

Latentish committed:
Minor aesthetic changes

-Swaps some consoles for the combo icons.
-Adds some floor decals to the engineer rooms.

Latentish committed:
De-souls the wiring

-The silliness is gone :sob:

Latentish committed:
replaces the ammo

-38 hunting was never meant to be apparently.

spockye committed:
Merge branch 'master' into New-Age-Ranger

spockye reviewed (approved):
Que the crazy frog music
